### ISSUE TITLE
Hide Taskbar icon if overlay is pinned

### ISSUE THREAD

**Issue by samfry13** [overlayeddev/overlayed#199]
**Is your feature request related to a problem? Please describe.**
On Windows, the overlay window stays present on the taskbar even when the overlay is pinned (i.e. hidden) and the user is not in a call

**Describe the solution you'd like**
If the user has the overlay pinned, don't show a window on the taskbar

**Describe alternatives you've considered**
N/A

**Additional context**
| Screenshot |
| - |
| ![image](https://github.com/user-attachments/assets/b7aedb79-5779-4b6b-abc5-1b0cf2a55ecc) |

Tauri has a setting for [skipping the taskbar](https://v2.tauri.app/reference/javascript/api/namespacewindow/#setskiptaskbar). This could potentially be set/unset when the user pins/unpins the overlay window respectively

**Comment by Hacksore** [comment]
Yeah I'll think we can support this but as an opt-in. We'd want something like this for the config:

- `windowsHideFromTaskbarWhenPinned` (for windows we can call [set_skip_taskbar](https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.set_skip_taskbar)
- `macHideFromDockWhenPinned` (for macOS we can call a custom impl to chage `set_activation_policy` to `Accessory`)
- Linux (no idea 😂)

@ahkohd can we change the activation policy dynamically at runtime so that when pinning we make it hidden from the dock and unpinning it goes back to normal?

**Comment by samfry13** [comment]
I believe `set_skip_taskbar` works for both windows and linux. It's just MacOS that isn't supported from that method.

**Comment by ahkohd** [comment]
> Yeah I'll think we can support this but as an opt-in. We'd want something like this for the config:
> 
>     * `windowsHideFromTaskbarWhenPinned` (for windows we can call [set_skip_taskbar](https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.set_skip_taskbar)
> 
>     * `macHideFromDockWhenPinned` (for macOS we can call a custom impl to chage `set_activation_policy` to `Accessory`)
> 
>     * Linux (no idea 😂)
> 
> 
> @ahkohd can we change the activation policy dynamically at runtime so that when pinning we make it hidden from the dock and unpinning it goes back to normal?
The activation policy is called in the Tauri setup function, which is already running at runtime. This should make it possible.

**Comment by ahkohd** [comment]
> Yeah I'll think we can support this but as an opt-in. We'd want something like this for the config:
> 
>     * `windowsHideFromTaskbarWhenPinned` (for windows we can call [set_skip_taskbar](https://docs.rs/tauri/latest/tauri/window/struct.Window.html#method.set_skip_taskbar)
> 
>     * `macHideFromDockWhenPinned` (for macOS we can call a custom impl to chage `set_activation_policy` to `Accessory`)
> 
>     * Linux (no idea 😂)
> 
> 
> @ahkohd can we change the activation policy dynamically at runtime so that when pinning we make it hidden from the dock and unpinning it goes back to normal?

The activation policy is called in the Tauri setup function, which is already running at runtime. This should make it possible.

**Comment by BlacRyu** [comment]
As it is I feel like there isn't much point to having a taskbar icon since it doesn't seem to serve any function, at least not on Windows. It doesn't seem to be possible to minimize the window, so you don't need to click on the icon to restore it. Clicking on the icon could maybe unpin the window to allow you to interact with it, but currently it doesn't do anything but give the invisible window focus. Right clicking on the icon and clicking 'Close Window' does nothing, nor does giving it focus and pressing alt+F4. 

I don't really see what function the task bar icon serves right now aside from indicating to the user that Overlayed is running, which the system tray icon does just as well along with many other more useful functions.

If made an option I would vote that the default option be to keep it hidden.

**Comment by francislavoie** [comment]
I actually think the opposite should be done, in a sense. I made changes in https://github.com/overlayeddev/overlayed/pull/266 relating to this; I think the settings window shouldn't be closed unless the app is closed (and should be opened but minimized by default) because it's particularly tricky to learn as a new user that the way to unpin is via the systray icon. I helped onboard a few streamers with it today and that was the first speedbump each of them encountered. So with the settings menu more easily discoverable, I added an unpin button in there.

**Comment by delCatta** [comment]
Mac/Windows user here (and just got Overlayed, thx guys!), would love to see the `macHideFromDockWhenPinned` option land. Once the overlay is pinned the Dock icon doesn't do much, the tray icon already handles quit / settings / unpin.
                                                                                              
I get @francislavoie's concern about unpinning being hard to discover for new users, but that feels like a separate UX problem from the icon living in the Dock. Could this be opt-in (default off) so new users still see it, and folks who know where the tray lives can hide it when they want? Happy to take a stab at it if no one's on it...  Let me know if you'd want a PR 🚀

**Comment by francislavoie** [comment]
I haven't had time to play with it some more in my PR #266 but I'm a Windows+Linux user so mac-specific stuff I would need help with. I'd gladly accept a PR on top of my branch if you want to base your changes off it.

**Comment by francislavoie** [comment]
I added another commit to my PR to add an option to hide the taskbar/dock icon when pinned (off by default).

**Comment by revunix** [comment]
Same issue on macOS:
When the overlay is pinned, the app still shows up permanently in the Dock. That’s pretty annoying and kind of defeats the purpose of pinning it.
Would be great if the Dock icon could also be hidden while the overlay is pinned.
